src/analyze/analyze-plot.c

```diff
@@ -391,5 +362,107 @@ int verb_plot(int argc, char *argv[], void *userdata) {
 
         svg("</svg>\n");
 
+        return 0;
+}
+
+static int show_table(Table* table, const char* word) {
+        int r;
+
+        assert(table);
+        assert(word);
+
+        if (table_get_rows(table) > 1) {
+                table_set_header(table, arg_legend);
+
+                if (!FLAGS_SET(arg_json_format_flags, JSON_FORMAT_OFF))
+                        r = table_print_json(table, NULL,
+                                             arg_json_format_flags |
+                                                 JSON_FORMAT_COLOR_AUTO);
+                else
+                        r = table_print(table, NULL);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to show table: %m");
+        }
+
+        if (arg_legend) {
+                if (table_get_rows(table) > 1)
+                        printf("\n%zu %s listed.\n", table_get_rows(table) - 1, word);
+                else
+                        printf("No %s.\n", word);
+        }
+
+        return 0;
+}
+
+static int produce_plot_as_text(UnitTimes* times) {
+        _cleanup_(table_unrefp) Table* table = NULL;
+        UnitTimes *u;
+        int r;
+
+        table = table_new("name", "activated", "activating", "time", "deactivated",
+                          "deactivating");
+        if (!table)
+                return log_oom();
+
+        for (u = times; u->has_data; u++) {
```

Comment:
we nowadays prefer if the iterator variable is declared inside the `for()` line, i.e..

```
for (UnitTimes *u = times; …
```